include/gsl/pointers

```diff
@@ -32,6 +32,15 @@
 namespace gsl
 {
 
+namespace details
+{
+template<typename T, typename = void>
+struct is_comparable_to_nullptr : std::false_type {};
+
+template <typename T>
+struct is_comparable_to_nullptr<T, std::is_convertible<decltype(std::declval<T>() != nullptr), bool>> : std::true_type {};
+}   // namespace details
```

Comment:
This specialization will only ever be matched if someone passes the type `std::is_convertible</*...stuff...*/>` as the second argument to `is_comparable_to_nullptr`. You need to arrange for the second parameter of your specialization to be `void`, but ideally a long and complicated spelling of `void` that's only valid when `std::declval<T>() != nullptr` is a valid expression that's convertible to `bool`. Something like:
```suggestion
template <typename T>
struct is_comparable_to_nullptr<T, std::enable_if_t<std::is_convertible<decltype(std::declval<T>() != nullptr), bool>::value>> : std::true_type {};
}   // namespace details
```
should work, for example.

Reply:
Thanks, that works. That was a silly error I made.